src/PolyType.SourceGenerator/Model/ConstructionWithComparer.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace PolyType.SourceGenerator.Model;
+
+public enum ConstructionWithComparer
+{
+    /// <summary>
+    /// No construction accepts any kind of comparer.
+    /// </summary>
+    None,
+
+    /// <summary>
+    /// The construction expects just the <see cref="IComparer{T}"/>.
+    /// </summary>
+    Comparer,
+
+    /// <summary>
+    /// The construction expects a <see cref="IComparer{T}"/> then the values.
+    /// </summary>
+    ComparerValues,
+
+    /// <summary>
+    /// The construction expects values and then a <see cref="IComparer{T}"/>.
+    /// </summary>
+    ValuesComparer,
+
+    /// <summary>
+    /// The construction expects just the <see cref="IEqualityComparer{T}"/>.
+    /// </summary>
+    EqualityComparer,
+
+    /// <summary>
+    /// The construction expects an <see cref="IEqualityComparer{T}"/> then the values.
+    /// </summary>
+    EqualityComparerValues,
+
+    /// <summary>
```

Comment:
Something I was discussing with @drewnoakes is we could avoid a combinatorial explosion of the various parameter permutations by modelling argument as an array of enums, i.e.
```C#
[IEqualityComparerOfT]
[IEnumerableOfT, IComparerOfT],
[SpanOfT, IEqualityComparerOfT]
```
etc.

Reply:
Yes, he did that. But he had an array of an array of enums. It was significantly more complicated than just a simple enum and I didn't see any reason for the complexity and extra allocations because I can't imagine what else we might add to this.